10-JavaScript/testing.md

```diff
@@ -23,96 +22,119 @@ In short, most interesting JavaScript code is full of *external dependencies*. M
+We will start writing tests with a sample whiteboarding problem.  
 
-1. Clone this repo: `https://github.com/AdaGold/scrabble-jasmine-baseline`
+1. Clone this repo: `https://github.com/AdaGold/pangram`
 1. `npm install`
 1. `npm start`
 
-Spend some time playing with the site. What models, collections and views do you think there might be? What functionality is still missing?
+Examine the README.  
 
-Now look through the code, and confirm or rebuke your suspicions. Pay close attention to the `spec` directory, which is where the tests live - we'll dive into it a little later. To run the tests, type `npm test` at the console.
+**Question**: In testing a Pangram, what edge-cases would you have?  
 
-Similar to Rails, the `spec` directory has the same structure as the `src` directory. Each file `src/dir/name.js` may have a corresponding `spec/dir/name.spec.js`.
+Now look through the code in the `pangram.spec.js` file, and see if the tests described match your predictions. To run the tests, type `npm test` at the console.
 
-Because we're running tests via `npm`, our tests have access to all our WebPack infrastructure, and `import` and `export` statements will work about as you'd expect.
+Similar to Rails, the `spec` directory has the same structure as the `src` directory. Each file `src/dir/name.js` may have a corresponding `spec/dir/name.spec.js`.
 
 ### Anatomy of a Test
 
-There are already some tests written for the `Word` model, so let's begin there. Open up `spec/models/word.spec.js`.
+There are already some tests written for the `isPangram` method, so let's begin there. Open up `spec/pangram.spec.js`.
 
-Because Jasmine is a behavior-driven testing language, testing using Jasmine for JavaScript is fairly similar to what we saw using Minitest for Ruby and Rails. As always, there are some specific differences.
+Because Jest is a behavior-driven testing language, testing using Jest for JavaScript is fairly similar to what we saw using Minitest for Ruby and Rails. As always, there are some specific differences.
 
 Each test should have these components to describe the test behavior.
 
 #### `describe` Blocks
-Describe blocks should create test groupings based on _objects_ and _functions_.
+Describe blocks should create test groupings based on _objects_ and _functions_.  They are optional, but provide some readability and organization.  Like Minitest you can also nest `describe` blocks.
 
-We should use one `describe` block for our overall `Word` model, and another nested one for each method or set of behavior we want to test. `Word` has one for the constructor, another for validations and a third for the `score()` method.
+We will use one `describe` block for our overall `Pangram` class, and another nested one for the `isPangram` method.
```

Comment:
Overall - I struggle with using this example since I think that having a method inside this class for this is really quite unnecessary. I can't really justify to the students why they would create an instance of a class with a string, just to call the method on this, rather than just create a method that takes a single string parameter. Do you feel like this is better than one of the Scrabble examples?

Reply:
Would it make more sense for you if we refactored the `Pangram` class to `TextBlock` or `Snippet` or something else that might have a `isPangram` method and implies there are other purposes for it?